### DISCUSSION TITLE
Error with Chapter 3, Part 7 code when pasting into Local Notebook

### DISCUSSION THREAD

**Discussion by anaconda121** [explosion/spaCy#8372] [Help: Coding & Implementations]
## How to reproduce the behaviour
I'm doing chapter 3, part 7 of the spacy course. The below code gets accepted on the course, but when I paste into my local notebook, it gives the error: `"TypeError: an integer is required"`. 

I tried changing the label from label = "ANIMAL" to label = nlp.vocab.strings("ANIMAL") as that worked earlier, but it didn't here. 

```py
import spacy
from spacy.matcher import PhraseMatcher
from spacy.tokens import Span

nlp = spacy.load("en_core_web_sm")
animals = ["Golden Retriever", "cat", "turtle", "Rattus norvegicus"]
animal_patterns = list(nlp.pipe(animals))
print("animal_patterns:", animal_patterns)
matcher = PhraseMatcher(nlp.vocab)
matcher.add("ANIMAL", None, *animal_patterns)

# Define the custom component
def animal_component(doc):
    # Apply the matcher to the doc
    matches = matcher(doc)
    # Create a Span for each match and assign the label "ANIMAL"
    spans = [Span(doc, start, end, label="ANIMAL") for match_id, start, end in matches]
    # Overwrite the doc.ents with the matched spans
    doc.ents = spans
    return doc


# Add the component to the pipeline after the "ner" component
nlp.add_pipe(animal_component, after="ner")
print(nlp.pipe_names)

# Process the text and print the text and label for the doc.ents
doc = nlp("I have a cat and a Golden Retriever")
print([(ent.text, ent.label_) for ent in doc.ents])
```

## Info about spaCy
* **spaCy version:** 2.0.16
* **Platform:** Windows-10-10.0.19041-SP0
* **Python version:** 3.7.10
* **Models:** en_core_web_md, en_core_web_sm

**Comment by polm** [comment]
This worked for me with a more recent version of spaCy v2. Can you try upgrading to 2.3.7 and running it again?

If you still get an error, please provide the whole stack trace, as from your sample code it's hard to tell where the error is occuring.
